packages/workspace/src/atom.ts

```diff
@@ -1,28 +1,54 @@
-import { isBrowser } from '@affine/env/constant';
-import type { WorkspaceFlavour } from '@affine/env/workspace';
-import type { WorkspaceVersion } from '@affine/env/workspace';
+import type { WorkspaceAdapter } from '@affine/env/workspace';
+import { WorkspaceFlavour, WorkspaceVersion } from '@affine/env/workspace';
 import type { EditorContainer } from '@blocksuite/editor';
+import { assertExists } from '@blocksuite/global/utils';
 import { atom } from 'jotai';
-import { atomWithStorage } from 'jotai/utils';
 import Router from 'next/router';
+import { z } from 'zod';
 
-export type RootWorkspaceMetadataV2 = {
-  id: string;
-  flavour: WorkspaceFlavour;
-  version: WorkspaceVersion;
-};
+const rootWorkspaceMetadataV1Schema = z.object({
+  id: z.string(),
+  flavour: z.nativeEnum(WorkspaceFlavour),
+});
 
-export type RootWorkspaceMetadataV1 = {
-  id: string;
-  flavour: WorkspaceFlavour;
-  // force type check
-  version: undefined;
-};
+const rootWorkspaceMetadataV2Schema = rootWorkspaceMetadataV1Schema.extend({
+  version: z.nativeEnum(WorkspaceVersion),
+});
+
+const rootWorkspaceMetadataArraySchema = z.array(
+  z.union([rootWorkspaceMetadataV1Schema, rootWorkspaceMetadataV2Schema])
+);
+
+export type RootWorkspaceMetadataV2 = z.infer<
+  typeof rootWorkspaceMetadataV2Schema
+>;
+
+export type RootWorkspaceMetadataV1 = z.infer<
+  typeof rootWorkspaceMetadataV1Schema
+>;
 
 export type RootWorkspaceMetadata =
   | RootWorkspaceMetadataV1
   | RootWorkspaceMetadataV2;
 
+export const workspaceAdaptersAtom = atom<
```

Comment:
`workspaceAdaptersAtom` is never changed. Why do we want it to be an atom?

Reply:
We haven’t but we will.

I might change it to a mock apis in the test environment.